packages/darklang/stdlib/string.dark

```diff
@@ -56,6 +106,68 @@ module Darklang =
       let fromChar (c: Char) : String = PACKAGE.Darklang.Stdlib.Char.toString c
 
 
+      /// URLBase64 encodes a string without padding. Uses URL-safe encoding with
+      /// {{-}} and {{_}} instead of {{+}} and {{/}}, as defined in
+      /// [RFC 4648 section 5](https://www.rfc-editor.org/rfc/rfc4648.html#section-5)
+      let base64Encode (s: String) : String = Builtin.String.base64Encode s
+
+
+      /// Base64 decodes a string. The returned value is wrapped in {{Result}}.
+      /// Works with both the URL-safe and standard Base64
+      /// alphabets defined in [RFC 4648
+      /// sections](https://www.rfc-editor.org/rfc/rfc4648.html)
+      /// [4](https://www.rfc-editor.org/rfc/rfc4648.html#section-4) and
+      /// [5](https://www.rfc-editor.org/rfc/rfc4648.html#section-5).
+      let base64Decode
+        (s: String)
+        : PACKAGE.Darklang.Stdlib.Result.Result<String, String> =
+        Builtin.String.base64Decode s
+
+
+      /// Take a string and hash it to a cryptographically-secure digest.
+      /// Don't rely on either the size or the algorithm.
+      let digest (s: String) : String = Builtin.String.digest s
+
+
+      /// Generate a <type String> of length <param length> from random characters
+      let random
+        (length: Int)
+        : PACKAGE.Darklang.Stdlib.Result.Result<String, String> =
+        Builtin.String.random length
+
+
+      /// Escape an untrusted string in order to include it safely in HTML output
+      let htmlEscape (html: String) : String =
+        html
+        |> Builtin.String.toList
+        |> PACKAGE.Darklang.Stdlib.List.map (fun c ->
+          match c with
+          | '<' -> "&lt;"
+          | '>' -> "&gt;"
+          | '&' -> "&amp;"
+          // include these for html-attribute-escaping
+          // even though they're not strictly necessary
+          // for html-escaping proper.
+          | '"' -> "&quot;"
+          // &apos; doesn't work in IE....
+          | ''' -> "&#x27;"
+          | _ -> PACKAGE.Darklang.Stdlib.String.fromChar c)
+        |> Builtin.String.join ""
+
+
+      /// Checks if <param lookingIn> contains <param searchingFor>
+      let contains (lookingIn: String) (searchingFor: String) : Bool =
+        match Builtin.String.indexOf lookingIn searchingFor with
+        | Some _ -> true
+        | None -> false
+
+
+      /// Returns the substring of <param string> between the <param from> and <param to> indices.
+      /// Negative indices start counting from the end of <param string>.
+      let slice (string: String) (from: Int) (``to``: Int) : String =
+        Builtin.String.slice_v0 string from ``to``
```

Comment:
It feels like the negative index stuff should be done directly here, vs in the builtin function.